pgdb/v1/message.go

```diff
@@ -6,13 +6,15 @@ import (
 
 type DBReflectMessage interface {
 	DBReflect() Message
+	DBReflectWithDialect(dialect Dialect) Message
```

Comment:
This is kinda tough... but an alternative isn't immediately jumping out at me.

I sort of think we might need to bite the bullet and require Dialect because 'falling back' to an old dialect if the caller neglects to specify is never permitted.